### PR TITLE
Metric SDK renames to align with draft SDK spec

### PR DESCRIPTION
**Pull request by jmacd**
Batcher is renamed Integrator
Ungrouped integrator is named `simple.Integrator`
SDK struct is renamed Accumulator.

There are no functional changes.

See https://github.com/open-telemetry/opentelemetry-specification/pull/347

### CODE REVIEW

File: sdk/metric/doc.go [hunk 1]
```diff
@@ -124,22 +124,22 @@ Aggregators implement a Merge method, also called in collection
...
+Integrator is an interface which sits between the SDK and an exporter.
+The Integrator embeds an AggregationSelector, used by the SDK to assign
+new Aggregators.  The Integrator supports a Process() API for submitting
+checkpointed aggregators to the integrator, and a CheckpointSet() API
 for producing a complete checkpoint for the exporter.  Two default
-Batcher implementations are provided, the "defaultkeys" Batcher groups
+Integrator implementations are provided, the "defaultkeys" Integrator groups
 aggregate metrics by their recommended Descriptor.Keys(), the
-"ungrouped" Batcher aggregates metrics at full dimensionality.
+"ungrouped" Integrator aggregates metrics at full dimensionality.
```
Comment (Aneurysm9) [inline sdk/metric/doc.go:134]:
```suggestion
"simple" Integrator aggregates metrics at full dimensionality.
```

File: sdk/metric/sdk.go [hunk 2]
```diff
@@ -301,32 +302,32 @@ func (s *syncInstrument) RecordOne(ctx context.Context, number api.Number, kvs [
...
+// The Accumulator does not start any background process to collect itself
+// periodically, this responsbility lies with the integrator, typically,
 // depending on the type of export.  For example, a pull-based
-// batcher will call Collect() when it receives a request to scrape
-// current metric values.  A push-based batcher should configure its
+// integrator will call Collect() when it receives a request to scrape
+// current metric values.  A push-based integrator should configure its
 // own periodic collection.
-func New(batcher export.Batcher, opts ...Option) *SDK {
+func New(integrator export.Integrator, opts ...Option) *Accumulator {
```
Comment (MrAlias) [inline sdk/metric/sdk.go:314]: Should this be renamed to `NewAccumulator`? It no longer is scoped by the package name (`sdk.New` doesn't return an `SDK` anymore).

Comment (jmacd) [reply sdk/metric/sdk.go]: Done.

File: sdk/metric/doc.go (hunk 1 above)
Comment (jmacd) [reply sdk/metric/doc.go]: Done.

**Pull request merged**